### PR TITLE
Initial push for publishing Driver Alert messages into a kafka stream

### PR DESCRIPTION
**Pull request by tonychen091**
Probably needs a little refactoring, but this feature currently pushes the following Json into a Driver Alert Kafka Topic

```{"alert":[68,78,80,87],"location":{"latitude":404739938,"longitude":-1049692256,"elevation":15070,"speed":51,"heading":26467},"status":"COMPLETE","bundleId":2,"step":0,"filename":"driverAlert_1493923473_fe80_3A_3A226_3Aadff_3Afe05_3A2561.csv","recordType":"driverAlert","utcTimeInSec":1507216177,"mSec":627,"validSignature":false,"length":4}```

### CODE REVIEW

**Comment by tonychen091** [comment]
@hmusavi We're still looking at that Alert message content that I have in the JSON, confirming that value that WyDOT is expecting.

**Comment by tonychen091** [comment]
Updated Sample Output:

```{"alert":"DNPW","location":{"latitude":404739938,"longitude":-1049692256,"elevation":15070,"speed":51,"heading":26467},"status":"COMPLETE","bundleId":2,"step":0,"filename":"driverAlert_1493923473_fe80_3A_3A226_3Aadff_3Afe05_3A2561.csv","recordType":"driverAlert","utcTimeInSec":1507216177,"mSec":627,"validSignature":false,"length":4}```

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java [hunk 1]
```diff
@@ -69,23 +65,32 @@ public void publish(BufferedInputStream bis, String fileName, ImporterFileType f
...
-      Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
-            EncodingRule.UPER);
-      metadata.addEncoding(msgEncoding).addEncoding(unsecuredDataEncoding);
-      OdeAsn1Data asn1Data = new OdeAsn1Data(metadata, payload);
 
-      publisher.publish(xmlUtils.toXml(asn1Data),
-         publisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
+      if (fileParser instanceof DriverAlertFileParser){
+         logger.debug("Publishing a driver alert.");
+         publisher.publish(JsonUtils.toJson(fileParser, false),
```
Comment (hmusavi) [inline jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java]: Architecturally, the ODE should always be sending out data in the consistent `{metadata:{}, payload:{}}` format. So I would recommend creating a `OdeDriverAlertPayload` and `OdeDriverAlertMetadata` and build it simliar to how it's done in the `else` part  and add `else` part to the existing `if (fileParser instanceof TimLogFileParser)`.

Comment (tonychen091) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java]: So for the data contained in Driver Alerts, there isn't really a defined payload vs metadata fields. Potentially we could populate fields that that make sense as metadata and put the alert as the payload, but the alerts are just a short code. Thoughts?

Comment (hmusavi) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java]: True, but nevertheless, some metadata fields are pretty standard such as `odeReceivedAt`, `serialId`, etc. We should also try to populate `recordGeneratedAt` and `recordGeneratedBy` if at all possible. Perhaps we could ask WYDOT to include a timestamp int he log file just like they do for BSMs. If can't do that, we should leave `recordGeneratedAt` blank but we could still populate `recordGeneratedBy`. The important thing is the consistency and uniformity of the ODE interface schema, architecturally speaking.

Comment (tonychen091) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java]: Agreed, we can make those edits. Got a preference as to where `alert` should go?

Comment (hmusavi) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java]: The `alert` itself will need to be the one and only element of a `OdeDriverAlert` class in `us.dot.its.jpo.ode.model` package extending `OdeObject`. `OdeDriverAlertPayload` and `OdeDriverAlertMetadata` should extend `OdeMsgPayload` and `OdeLogMetadata`, respectively. `OdeDriverAlert` is then set as the `data` element of `OdeDriverAlertPayload`. Model them after `OdeBsmPayload` and `OdeBsmMetadata`. Doing so will populate much of the common fields automatically if `OdeDriverAlertMetadata` is instantiated using the constructor that accepts a `OdeDriverAlertPayload`. So you will instantiate the payload first and then the metadata object given the payload.

**Comment by tonychen091** [comment]
@hmusavi Let me know what you think about my latest commits, I tried to follow the previous architecture a little better. Moved `alert` into the payload portion of the data object, as I think that keeps it inline with the other objects better. We could move it into OdeDriveralertData but would that serialize properly?

**Pull request merged**